src/sentry/api/endpoints/team_project_index.py

```diff
@@ -136,3 +148,22 @@ def post(self, request, team):
 
             return Response(serialize(project, request.user), status=201)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def put(self, request, team):
+        if not request.user.is_authenticated():
+            raise PermissionDenied(detail="Key doesn't have permission to edit Project")
+        try:
+            project = Project.objects.get(
+                slug=request.DATA['slug']
+            )
+        except Project.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        serializer = ProjectSerializer(data=request.DATA, partial=True)
```

Comment:
if we use the base serializer here we should probably support the rest of the attributes as otherwise it just fails quietly 

one option would be to simply put this under ProjectDetailsEndpoint (probably better), and not allow you to PUT via this endpoint. The other option would be to passthru all options to that endpoint (more complex)